src/codegen_internal.h

```diff
@@ -1,6 +1,6 @@
 // This file is a part of Julia. License is MIT: http://julialang.org/license
 
-#if defined(LLVM38) && !defined(LLVM37)
+#if JL_LLVM_VERSION >= 30800
```

Comment:
Which is why I said,

> The condition there before was false

This was added before the JIT and codegen are split and therefore before `jitlayers.h` exists. This is needed only by `jitlayers.cpp`, which doesn't include `codegen_internal.h`, for LLVM 3.7.x which is why it should be moved to `jitlayers.h`.